scripts/run_all.py

```diff
@@ -49,41 +125,91 @@ def run_build() -> None:
+        # start all examples in parallel
+        for path in examples:
+            # each example gets its own viewer
+            viewer = Viewer(web=True).start()
+            example = run_py_example(path, viewer_port=viewer.port, wait=False)
+            processes.append((viewer, example))
+        # wait for all processes to finish, and close the viewers
+        for process in processes:
+            viewer, example = process
+            example.wait()
+            viewer.close()
+    else:
+        with Viewer(close=True, web=True) as viewer:
+            for example in examples:
+                run_py_example(example, viewer_port=viewer.port)
+
+
+def run_separate_with(examples: List[str], close: bool, start_example: Callable[[str, int, bool], Any]) -> None:
+    processes: List[Tuple[Any, Any]] = []
+    # start all examples in parallel
+    for path in examples:
+        # each example gets its own viewer
+        viewer = Viewer().start()
+        example = start_example(path, viewer.port, False)
+        processes.append((viewer, example))
+    # wait for all examples to finish
+    for process in processes:
+        process[1].wait()
+    # close viewers if requested
+    if close:
+        for process in processes:
+            process[0].close()
+
+
+def run_save(examples: List[str], separate: bool, close: bool) -> None:
+    with Viewer(close=True) as viewer:  # ephemeral viewer that exists only while saving
+        for example in examples:
+            run_py_example(example, viewer_port=viewer.port, save="out.rrd")
+
+    if separate:
+        run_separate_with(examples, close, lambda path, viewer_port, wait: run_saved_example(path, wait))
+    else:
+        with Viewer(close) as viewer:
+            for example in examples:
+                run_saved_example(example)
+
+
+def run_native(examples: List[str], separate: bool, close: bool) -> None:
+    if separate:
+        run_separate_with(examples, close, run_py_example)
+    else:
+        with Viewer(close) as viewer:
+            for example in examples:
+                run_py_example(example, viewer_port=viewer.port)
+
+
 def main() -> None:
     parser = argparse.ArgumentParser(description="Runs all examples.")
-    parser.add_argument("--skip-build", action="store_true", help="Skip building the Python SDK")
-    parser.add_argument("--web", action="store_true", help="Run all examples in a web viewer.")
+    parser.add_argument("--skip-build", action="store_true", help="Skip building the Python SDK.")
+    parser.add_argument("--web", action="store_true", help="Run examples in a web viewer.")
     parser.add_argument(
         "--save",
         action="store_true",
-        help="Run all examples, save them to disk as rrd, then view them natively.",
+        help="Run examples, save them to disk as rrd, then view them natively.",
     )
+    parser.add_argument("--fast", action="store_true", help="Run only examples which complete quickly.")
```

Comment:
The issue also mentions:
> Run different sub-sets of the examples ("all", "fast", "tests", …)
What does `tests` refer to?

Reply:
Nothing yet, but we have plans to move some examples (`api_demo`) to a separate `test` folder, but we haven't gotten to it yet.